polyfills/Element/prototype/dataset/detect.js

```diff
@@ -0,0 +1,6 @@
+(document.documentElement.dataset &&
+	(
+		Object.getOwnPropertyDescriptor(Element.prototype, 'dataset')  &&
+		Object.getOwnPropertyDescriptor(Element.prototype, 'dataset').get
```

Comment:
We could use `__getter__` if `getOwnPropertyDescriptor` isn't there.
I've included `getOwnPropertyDescriptor` in the dependencies which should mean that it is present for the detect.